crates/egui_extras/src/loaders/gif_loader.rs

```diff
@@ -0,0 +1,135 @@
+use egui::{
+    ahash::HashMap,
+    decode_gif_uri,
+    load::{Bytes, BytesPoll, ImageLoadResult, ImageLoader, ImagePoll, LoadError, SizeHint},
+    mutex::Mutex,
+    ColorImage, GifFrameDurations, Id,
+};
+use image::AnimationDecoder as _;
+use std::{io::Cursor, mem::size_of, sync::Arc, time::Duration};
+
+/// Array of Frames and the duration for how long each frame should be shown
+#[derive(Debug, Clone)]
+pub struct AnimatedImage {
+    frames: Vec<Arc<ColorImage>>,
+    frame_durations: GifFrameDurations,
+}
+
+impl AnimatedImage {
+    pub fn byte_len(&self) -> usize {
+        size_of::<Self>()
+            + self
+                .frames
+                .iter()
+                .map(|image| {
+                    image.pixels.len() * size_of::<egui::Color32>() + size_of::<Duration>()
+                })
+                .sum::<usize>()
+    }
+
+    /// Gets image at index
+    pub fn get_image(&self, index: usize) -> Arc<ColorImage> {
+        self.frames[index % self.frames.len()].clone()
+    }
+}
+type Entry = Result<Arc<AnimatedImage>, String>;
+
+#[derive(Default)]
+pub struct GifLoader {
+    cache: Mutex<HashMap<String, Entry>>,
+}
+
+impl GifLoader {
+    pub const ID: &'static str = egui::generate_loader_id!(GifLoader);
+}
+
+pub fn gif_to_sources(data: Bytes) -> Result<AnimatedImage, String> {
+    let decoder = image::codecs::gif::GifDecoder::new(Cursor::new(data))
+        .map_err(|_err| "Couldnt decode gif".to_owned())?;
+    let mut images = vec![];
+    let mut durations = vec![];
+    for frame in decoder.into_frames() {
+        let frame = frame.map_err(|_err| "Couldnt decode gif".to_owned())?;
```

Comment:
Don't throw away an error message

```suggestion
        let frame = frame.map_err(|err| format!("Failed to decode gif: {err}"))?;
```